RETE-14

seanans opened the pull request:
-Create Post
-Get All Posts
-Get Post By ID

seanans committed:
-Create Post
-Get All Posts
-Get Post By ID

BobMakhlin commented on src/main/java/com/laborete/LaboreteAPI/posts/controller/PostsController.java:
```diff
@@ -0,0 +1,55 @@
+        description = "Posts Controller"
+)
+public class PostsController {
+    @Autowired
```
You inejct `PostsService` via constructor bellow. Therefore there is no need for `@Autowired` here. Decide between `@Autowired` and constructor injection.

BobMakhlin commented on src/main/java/com/laborete/LaboreteAPI/posts/entity/PostEntity.java:
```diff
@@ -0,0 +1,48 @@
+//
```
Remove this garbage comment

BobMakhlin commented on src/main/java/com/laborete/LaboreteAPI/posts/exception/ResourceBadRequestException.java:
```diff
@@ -0,0 +1,18 @@
+package com.laborete.LaboreteAPI.posts.exception;
+
+public class ResourceBadRequestException extends RuntimeException {
```
Code duplication

BobMakhlin commented on src/main/java/com/laborete/LaboreteAPI/posts/exception/ResourceNotFoundException.java:
```diff
@@ -0,0 +1,18 @@
+package com.laborete.LaboreteAPI.posts.exception;
+
+public class ResourceNotFoundException extends RuntimeException {
```
Code duplication

BobMakhlin commented on src/main/java/com/laborete/LaboreteAPI/posts/models/CreatePostDTO.java:
```diff
@@ -0,0 +1,24 @@
+public class CreatePostDTO {
+
+    @JsonProperty("id")
+    private UUID id;
```
Id is an internal field. When you create something you should not pass any id - it has to be generated under the hood.

BobMakhlin commented on src/main/java/com/laborete/LaboreteAPI/posts/models/PostDTO.java:
```diff
@@ -0,0 +1,32 @@
+import java.util.UUID;
+@Getter
+@Setter
+public class PostDTO {
```
Dto does not follow the required structure (see the ticket)

BobMakhlin commented on src/main/java/com/laborete/LaboreteAPI/posts/service/PostsService.java:
```diff
@@ -0,0 +1,15 @@
+public interface PostsService {
+    List<PostDTO> getAllPosts();
+
+    PostDTO createPost(CreatePostDTO user);
```
user? :)

BobMakhlin commented on src/main/java/com/laborete/LaboreteAPI/posts/service/PostsServiceImpl.java:
```diff
@@ -0,0 +1,99 @@
+        List<PostEntity> postEntities = this.postRepository.findAll();
+        List<PostDTO> postDTOList = new ArrayList();
+        for (PostEntity postEntity : postEntities) {
+            PostDTO postDTO = this.postMapper.postEntityToPostDTO(postEntity);
```
Mapper is able to map collections! No need for this `for` loop.

BobMakhlin commented on src/main/java/com/laborete/LaboreteAPI/posts/service/PostsServiceImpl.java:
```diff
@@ -0,0 +1,99 @@
+    }
+
+
+    public PostDTO findById(UUID id) {
```
There was a good method naming convension. 
`getPostById` -> returns a post if it's found; otherwise throws.
`findPostById` -> returns a post if it's found; otherwise returns null (in java there is an alternative to this approach - `Optional`).

BobMakhlin commented on src/main/java/com/laborete/LaboreteAPI/posts/models/PostDTO.java:
```diff
@@ -0,0 +1,32 @@
+    @JsonProperty("userPosition")
+    private String userPosition;
+    @JsonProperty("userAvatarEntity")
+    private UserAvatarDTO userAvatarDTO;
```
Please get rid of avatar by now. We will introduce it a bit later. But thanks for your commitment!

seanans committed:
Pull Fix
-Changed names
-Deleted garbage comments
-Cleaned injection with constructor in PostController
-PostServiceImpl refactored
